### ISSUE TITLE
Clarify in-cluster documentation

### ISSUE THREAD

**Issue by alanraison** [aws/amazon-eks-pod-identity-webhook#153]
**What would you like to be added**:

The main readme starts:

> After version v0.3.0, `--in-cluster=true` no longer works and is deprecated. Please use `--in-cluster=false` and manage the cluster certificate with cert-manager or some other external certificate provisioning system. This is because certificates using the legacy-unknown signer are no longer signed when using the v1 certificates API.

I am trying to run the webhook in my own cluster and have followed the SELF_HOSTING guidance, which recommends setting kube-apiserver configuration for keys and following the in cluster installation method. I also note that I am using certificate manager to hold a certificate, whose values are likely different to those in the kube-apiserver configuration, though perhaps this is for a different purpose?

I also get the following log message upon startup:

```
Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
```

"--master" does not appear in the latest branch or in the readme,

In summary I would like some clarification on whether this webhook can be run inside a self-hosted cluster and how this should work.

**Why is this needed**:

This is needed to clear up the contradictions in the main readme vs the Self Hosted readme.

I am using version 0d254ee of the webhook.

**Comment by alanraison** [comment]
I have found that the log message comes from client-go, so is unrelated and unconcerning.
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/tools/clientcmd/client_config.go#L617

**Comment by alanraison** [comment]
I have now succeeded in configuring this. The inCluster documentation was ok, but a little confusing when compared to the go-client terminology.
